### PR TITLE
[Notifier] [OvhCloud] Add tests

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.1
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | ---
| License       | MIT
| Doc PR        | ---

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/OvhCloud/OvhCloudTransportFactory.php [hunk 1]
```diff
@@ -29,7 +30,17 @@ public function create(Dsn $dsn): TransportInterface
...
+        if (!$consumerKey) {
+            throw new IncompleteDsnException('Missing consumer_key.', $dsn->getOriginalDsn());
+        }
+
         $serviceName = $dsn->getOption('service_name');
+
+        if (!$serviceName) {
+            throw new IncompleteDsnException('Missing service_name.', $dsn->getOriginalDsn());
+        }
+
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/OvhCloud/OvhCloudTransportFactory.php]: This should be considered a bugfix

Comment (derrabus) [reply src/Symfony/Component/Notifier/Bridge/OvhCloud/OvhCloudTransportFactory.php]: see https://github.com/symfony/symfony/pull/39406#discussion_r539307754

File: src/Symfony/Component/Notifier/Bridge/OvhCloud/composer.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "require": {
         "php": ">=7.2.5",
         "symfony/http-client": "^4.3|^5.0",
-        "symfony/notifier": "^5.0,<5.2"
+        "symfony/notifier": "^5.1,<5.2"
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/OvhCloud/composer.json]:
```suggestion
        "symfony/notifier": "~5.1.0"
```

File: src/Symfony/Component/Notifier/Bridge/OvhCloud/Tests/OvhCloudTransportFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        $factory = $this->createFactory();
+
+        $dsn = 'ovhcloud://applicationKey:applicationSecret@default?consumer_key=consumerKey&service_name=serviceName';
+        $transport = $factory->create(Dsn::fromString($dsn));
+        $transport->setHost('host.test');
+
+        $this->assertSame('ovhcloud://host.test?consumer_key=consumerKey&service_name=serviceName', (string) $transport);
+    }
+
+    public function testCreateWithMissingOptionConsumerKeyFromThrowsIncompleteDsnException()
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/OvhCloud/Tests/OvhCloudTransportFactoryTest.php]:
```suggestion
    public function testCreateWithMissingOptionConsumerKeyThrowsIncompleteDsnException()
```

File: src/Symfony/Component/Notifier/Bridge/OvhCloud/Tests/OvhCloudTransportFactoryTest.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        $factory = $this->createFactory();
+
+        $this->expectException(IncompleteDsnException::class);
+
+        $dsnIncomplete = 'ovhcloud://applicationKey:applicationSecret@default?service_name=serviceName';
+        $factory->create(Dsn::fromString($dsnIncomplete));
+    }
+
+    public function testCreateWithMissingOptionServiceNameFromThrowsIncompleteDsnException()
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/OvhCloud/Tests/OvhCloudTransportFactoryTest.php]:
```suggestion
    public function testCreateWithMissingOptionServiceNameThrowsIncompleteDsnException()
```

**Comment by nicolas-grekas** [comment]
Thank you @OskarStark.

**Pull request merged**